apps/server/src/modules/group/uc/group.uc.ts

```diff
@@ -57,32 +60,59 @@ export class GroupUc {
 		return page;
 	}
 
-	private async findCombinedClassListForSchool(schoolId: string): Promise<ClassInfoDto[]> {
-		const [classInfosFromClasses, classInfosFromGroups] = await Promise.all([
-			await this.findClassesForSchool(schoolId),
-			await this.findGroupsOfTypeClassForSchool(schoolId),
-		]);
+	private async findCombinedClassListForSchool(
+		schoolId: string,
+		schoolYearQueryType: SchoolYearQueryType | undefined
+	): Promise<ClassInfoDto[]> {
+		let classInfosFromGroups: ClassInfoDto[] = [];
+
+		const classInfosFromClasses = await this.findClassesForSchool(schoolId, schoolYearQueryType);
+
+		if (schoolYearQueryType === SchoolYearQueryType.CURRENT_YEAR || schoolYearQueryType === undefined) {
+			classInfosFromGroups = await this.findGroupsOfTypeClassForSchool(schoolId);
+		}
 
 		const combinedClassInfo: ClassInfoDto[] = [...classInfosFromClasses, ...classInfosFromGroups];
 
 		return combinedClassInfo;
 	}
 
-	private async findClassesForSchool(schoolId: EntityId): Promise<ClassInfoDto[]> {
+	private async findClassesForSchool(
+		schoolId: EntityId,
+		schoolYearQueryType: SchoolYearQueryType | undefined
+	): Promise<ClassInfoDto[]> {
 		const classes: Class[] = await this.classService.findClassesForSchool(schoolId);
+		const currentYear: SchoolYearEntity = await this.schoolYearService.getCurrentSchoolYear();
 
-		const classInfosFromClasses: ClassInfoDto[] = await Promise.all(
-			classes.map(async (clazz: Class): Promise<ClassInfoDto> => {
-				const teachers: UserDO[] = await Promise.all(
-					clazz.teacherIds.map((teacherId: EntityId) => this.userService.findById(teacherId))
-				);
-
+		const classesWithSchoolYear: { clazz: Class; schoolYear?: SchoolYearEntity }[] = await Promise.all(
+			classes.map(async (clazz) => {
 				let schoolYear: SchoolYearEntity | undefined;
 				if (clazz.year) {
 					schoolYear = await this.schoolYearService.findById(clazz.year);
 				}
 
-				const mapped: ClassInfoDto = GroupUcMapper.mapClassToClassInfoDto(clazz, teachers, schoolYear);
+				return {
+					clazz,
+					schoolYear,
+				};
+			})
+		);
+
+		const filteredClassesForSchoolYear = classesWithSchoolYear.filter((classWithSchoolYear) =>
+			this.isClassOfQueryType(currentYear, classWithSchoolYear.schoolYear, schoolYearQueryType)
+		);
+
+		const classInfosFromClasses = await Promise.all(
```

Comment:
You can devide this function up to enhance the readability. Actually it looks more complicated than it has to be